### ISSUE TITLE
[@typescript-eslint/type-annotation-spacing] false positive

### ISSUE THREAD

**Issue by crfrolik** [typescript-eslint/typescript-eslint#1667]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/type-annotation-spacing": "error",
  }
}
```

```TS
type Constructor<T> = new (...args: any[]) => T;
```

**Expected Result**

No lint error.

**Actual Result**

```
  25:44  error  Unexpected space before the '=>'  @typescript-eslint/type-annotation-spacing

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

Using the `--fix` option produces the following:

```TS
type Constructor<T> = new (...args: any[])=> T;
```

**Additional Info**

This was a regression in v2.22.0. It did not occur in previous versions.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.8.0` |
| `yarn`                              | `1.22.0` |

**Comment by bradzacher** [comment]
Please use the search next time.
This has been raised and fixed.

https://github.com/typescript-eslint/typescript-eslint/issues/1663
